test/fw/ptl/utils/plugins/ptl_test_runner.py

```diff
@@ -734,16 +750,19 @@ def __are_requirements_matching(param_dic=None, test=None):
             elif eff_tc_req['min_server_disk'] >= available_sys_disk:
                 return False
         if set(param_dic['moms']) & set(param_dic['servers']):
-            if eff_tc_req['no_mom_on_server']:
+            if eff_tc_req['no_mom_on_server'] or \
+               (set(_nomom) - set(_servers)) or \
+               _no_mom_on_server:
                 return False
         else:
```

Comment:
Now that I think about it, do we even need this else? Is it really necessary?

Reply:
That's right Vincent. I thought the same. Removed else condition and it's working fine without else.